src/video/windows/SDL_windowskeyboard.c

```diff
@@ -54,33 +54,58 @@ static void IME_SetTextInputArea(SDL_VideoData *videodata, HWND hwnd, const SDL_
 #define MAPVK_VSC_TO_VK 1
 #endif
 
-// Alphabetic scancodes for PC keyboards
-void WIN_InitKeyboard(SDL_VideoDevice *_this)
+#define MAX_KEYMAP_PRECACHE 4
```

Comment:
I'm tempted to lower this. Most people will have one or two layouts that will be active at any given time. What if we just precache the current layout?